qa/rpc-tests/fundrawtransaction.py

```diff
@@ -643,5 +643,123 @@ def run_test(self):
+
+        # Make sure there is exactly one input so coin selection can't skew the result
+        assert_equal(len(self.nodes[3].listunspent(1)), 1)
+
+        inputs = []
+        addr1 = self.nodes[2].getnewaddress()
+        addr2 = self.nodes[2].getnewaddress()
+        outputs = {addr1: 1}
+        rawtx = self.nodes[3].createrawtransaction(inputs, outputs)
+
+        result = [self.nodes[3].fundrawtransaction(rawtx), # uses min_relay_tx_fee (set by settxfee)
+                  self.nodes[3].fundrawtransaction(rawtx, {"subtractFeeFromAmount": []}), # empty subtraction list
+                  self.nodes[3].fundrawtransaction(rawtx, {"subtractFeeFromAmount": [addr2]}), # addr2 isn't an output, so no subtraction
+                  self.nodes[3].fundrawtransaction(rawtx, {"subtractFeeFromAmount": [addr1]}), # uses min_relay_tx_fee (set by settxfee)
+                  self.nodes[3].fundrawtransaction(rawtx, {"feeRate": 2*min_relay_tx_fee}),
+                  self.nodes[3].fundrawtransaction(rawtx, {"feeRate": 2*min_relay_tx_fee, "subtractFeeFromAmount": [addr1]})]
+
+        dec_tx = [self.nodes[3].decoderawtransaction(result[0]['hex']),
+                  self.nodes[3].decoderawtransaction(result[1]['hex']),
+                  self.nodes[3].decoderawtransaction(result[2]['hex']),
+                  self.nodes[3].decoderawtransaction(result[3]['hex']),
+                  self.nodes[3].decoderawtransaction(result[4]['hex']),
+                  self.nodes[3].decoderawtransaction(result[5]['hex'])]
+
+        output = [dec_tx[0]['vout'][1 - result[0]['changepos']]['value'],
+                  dec_tx[1]['vout'][1 - result[1]['changepos']]['value'],
+                  dec_tx[2]['vout'][1 - result[2]['changepos']]['value'],
+                  dec_tx[3]['vout'][1 - result[3]['changepos']]['value'],
+                  dec_tx[4]['vout'][1 - result[4]['changepos']]['value'],
+                  dec_tx[5]['vout'][1 - result[5]['changepos']]['value']]
+
+        change = [dec_tx[0]['vout'][result[0]['changepos']]['value'],
+                  dec_tx[1]['vout'][result[1]['changepos']]['value'],
+                  dec_tx[2]['vout'][result[2]['changepos']]['value'],
+                  dec_tx[3]['vout'][result[3]['changepos']]['value'],
+                  dec_tx[4]['vout'][result[4]['changepos']]['value'],
+                  dec_tx[5]['vout'][result[5]['changepos']]['value']]
+
+        # total fee isn't affected by the subtractFeeFromAmount option
+        assert(result[0]['fee'] == result[1]['fee'] == result[2]['fee'] == result[3]['fee'])
+        assert(result[4]['fee'] == result[5]['fee'])
+
+        # change amounts in result 0, 1, and 2 are the same
+        assert(change[0] == change[1] == change[2])
+
+        # outputs in result 0, 1, and 2 are the same
+        assert(output[0] == output[1] == output[2])
+
+        # 0's output should be equal to 3's (output plus fee)
+        assert(output[0] == output[3] + result[3]['fee'])
```

Comment:
It appears that `assert_equal` can only compare two things. For cases like `assert(A == B == C == D)` would you prefer 3 separate `assert_equal()` calls instead?